src/main/java/qna/domain/Answers.java

```diff
@@ -0,0 +1,24 @@
+package qna.domain;
+
+import qna.CannotDeleteException;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+
+public class Answers {
+    private final List<Answer> answers;
+
+    public Answers(List<Answer> answers) {
+        this.answers = Collections.unmodifiableList(answers);
+    }
+
+    public List<DeleteHistory> delete(User loginUser, List<DeleteHistory> deleteHistories) throws CannotDeleteException {
```

Comment:
가능하면 인자로 전달 받은 값을 변경하기 보다는 새로운 값을 만들어 리턴하는게 좋습니다.
지금은 `deleteHistories` 가 List 자료구조여서 크게 와닿지 않을 수 있지만, 일종의 setter를 사용하는 것과 같다고 생각해요.
그리고 인자로 전달 받은 값에 문제가 생겼을 때 디버깅도 어렵구요.

아래와 같은 구조로 변경해보면 어떨까요?
```suggestion
    public List<DeleteHistory> delete(User loginUser) throws CannotDeleteException {
        List<DeleteHistory> deleteHistories = new ArrayList<>();
        // ...
        return deleteHistories;
    }    
```